File: packages/ts/tracing/src/debug-mode.ts
let debugModeEnabled = false;

export function enableDebugMode(): void {
  debugModeEnabled = true;
}

export function disableDebugMode(): void {
  debugModeEnabled = false;
}

export function isDebugModeEnabled(): boolean {
  return debugModeEnabled || process.env["LEMMA_DEBUG"] === "true";
}

export function lemmaDebug(prefix: string, msg: string, data?: Record<string, unknown>): void {
  if (!isDebugModeEnabled()) return;
  if (data !== undefined) {
    console.log(`[LEMMA:${prefix}] ${msg}`, data);
  } else {
    console.log(`[LEMMA:${prefix}] ${msg}`);
  }
}

File: packages/py/tracing/src/uselemma_tracing/debug_mode.py
from __future__ import annotations

import os

_debug_mode_enabled = False


def enable_debug_mode() -> None:
    global _debug_mode_enabled
    _debug_mode_enabled = True


def disable_debug_mode() -> None:
    global _debug_mode_enabled
    _debug_mode_enabled = False


def is_debug_mode_enabled() -> bool:
    return _debug_mode_enabled or os.environ.get("LEMMA_DEBUG") == "true"


def _lemma_debug(prefix: str, msg: str, **data: object) -> None:
    if not is_debug_mode_enabled():
        return
    if data:
        print(f"[lemma:{prefix}] {msg}", data)
    else:
        print(f"[lemma:{prefix}] {msg}")
